src/pages/admin/ClubAdminEmails.tsx

```diff
@@ -0,0 +1,218 @@
+import { useSnackbar } from "notistack";
+import { useContext, useEffect, useState } from "react";
+import { supabase } from "../../supabaseClient";
+import { Box, TextField, Typography } from "@mui/material";
+import AsyncButton from "../../comps/ui/AsyncButton";
+import UserContext from "../../comps/context/UserContext";
+
+const ClubAdminEmails = () => {
+    const { enqueueSnackbar } = useSnackbar();
+
+    const user = useContext(UserContext);
+
+    const [orgId, setOrgId] = useState<number>();
+    const [orgName, setOrgName] = useState("");
+    const [searchInput, setSearchInput] = useState("");
+    const [filteredOrgs, setFilteredOrgs] = useState<Organization[]>([]);
+    const [allOrgs, setAllOrgs] = useState<Organization[]>([]);
+    const [currentOrganization, setCurrentOrganization] =
+        useState<Organization>();
+    const [adminEmails, setAdminEmails] = useState<string>("");
+
+    useEffect(() => {
+        const fetchAllOrgs = async () => {
+            const { data, error } = await supabase
+                .from("organizations")
+                .select("*");
+            if (error || !data) {
+                enqueueSnackbar(
+                    "Failed to load organizations. Contact it@stuysu.org for support.",
+                    { variant: "error" },
+                );
+                return;
+            }
+            setAllOrgs(data);
+        };
+
+        fetchAllOrgs();
+    }, [enqueueSnackbar]);
+
+    useEffect(() => {
+        if (searchInput) {
+            setFilteredOrgs(
+                allOrgs.filter((org) =>
+                    org.name.toLowerCase().includes(searchInput.toLowerCase()),
+                ),
+            );
+        } else {
+            setFilteredOrgs([]);
+        }
+    }, [searchInput, allOrgs]);
+
+    // useEffect(() => {
```

Comment:
Probably should remove commented code before merging, unless you believe it should be used later